### PR TITLE
Add nix template

### PR DESCRIPTION
**Pull request by KaiserKarel**
Uses GA templating to remove a lot of boilerplate.

### CODE REVIEW

**Review by hussein-aitlahcen — APPROVED** [review]

**Review by cor — APPROVED** [review]

File: .github/workflows/build-test-deploy.yml [hunk 1]
```diff
@@ -576,63 +400,34 @@ jobs:
...
+  #   strategy:
+  #     matrix:
+  #       runtime: [picasso, composable]
+  #   container:
+  #     image: niteo/nixpkgs-nixos-22.05:316b762afdb9e142a803f29c49a88b4a47db80ee
+  #   steps:
+  #     - uses: actions/checkout@v3
+  #     - uses: "./.github/templates/watch-exec"
+  #       with:
+  #         command: nix -- run .#simnode-tests-${{ matrix.runtime }}
```
Comment (dzmitry-lahoda) [inline .github/workflows/build-test-deploy.yml:429]: it misses google bucket cp usage as I see. so it starts from empty DB. from genesis. which was migrated. which will fail.

Comment (KaiserKarel) [reply .github/workflows/build-test-deploy.yml:429]: Ah gotcha

Comment (KaiserKarel) [reply .github/workflows/build-test-deploy.yml:429]: I'll re-enable

**Comment by KaiserKarel** [comment]
@Mergifyio unqueue

File: .github/workflows/build-test-deploy.yml [hunk 2]
```diff
@@ -610,29 +423,20 @@ jobs:
...
-          unzip -o "$FILENAME" -d /tmp/db
-          cachix watch-exec -j 16 -c 0 composable-community nix -- run .#simnode-tests-${{ matrix.runtime }} --no-update-lock-file --show-trace -L
+      - uses: "./.github/templates/watch-exec"
+        with:
+          pre-run: |
+            FILENAME=$(date -d yesterday +'%m-%d-%Y').zip
+            GS_BUCKET="${{ matrix.runtime }}-data-store"
+            gsutil cp gs://$GS_BUCKET/"$FILENAME" .
+            unzip -o "$FILENAME" -d /tmp/db
+          command: nix -- run .#simnode-tests-${{ matrix.runtime }}
```
Comment (dzmitry-lahoda) [inline .github/workflows/build-test-deploy.yml:437]: thans for simnode back:)

Comment (dzmitry-lahoda) [inline .github/workflows/build-test-deploy.yml:437]: buildCommand

File: .github/workflows/build-test-deploy.yml [hunk 3]
```diff
@@ -610,29 +423,20 @@ jobs:
...
           export_default_credentials: true
-      - run: |
-          FILENAME=$(date -d yesterday +'%m-%d-%Y').zip
-          GS_BUCKET="${{ matrix.runtime }}-data-store"
-          gsutil cp gs://$GS_BUCKET/"$FILENAME" .
-          unzip -o "$FILENAME" -d /tmp/db
-          cachix watch-exec -j 16 -c 0 composable-community nix -- run .#simnode-tests-${{ matrix.runtime }} --no-update-lock-file --show-trace -L
+      - uses: "./.github/templates/watch-exec"
+        with:
+          pre-run: |
```
Comment (dzmitry-lahoda) [inline .github/workflows/build-test-deploy.yml:432]: oh, no actions are look like nix derivations:)

**Review by dzmitry-lahoda — APPROVED** [review]

File: .github/workflows/build-test-deploy.yml (hunk 2 above)
Comment (KaiserKarel) [reply .github/workflows/build-test-deploy.yml:437]: 🖖

File: .github/workflows/build-test-deploy.yml (hunk 3 above)
Comment (KaiserKarel) [reply .github/workflows/build-test-deploy.yml:432]: we can definitely optimize more and reduce boilerplate, but I think this already helps out a lot.

**Comment by KaiserKarel** [comment]
@mergifyio requeue

**Pull request merged**